spec/jobs/follows/create_chat_channel_job_spec.rb

```diff
@@ -10,6 +10,12 @@
 
     it "creates a chat channel when mutual followers" do
       follow2 = create(:follow, follower: user2, followable: user)
+
+      # Follow has an after_create callback that creates a channel between the two users,
+      # so to make sure this test is correct, we delete all channels right after
+      ChatChannelMembership.delete_all
```

Comment:
As I specified in the comment, the `Follow` model has an `after_create` callback whose side effect is to create a `ChatChannel`. The job doesn't create a new channel if there's an existing one. I honestly have no idea what was the combination and the sequence of test leading to no channels so that this one would pass but by removing channels explicitly we can ensure the test always passes

Reply:
The callback contains the call to the job `perform_later`, which won't execute if the `queue_adapter` is set to test, so the channel was not actually created in the callback and the test passed. 
However, I agree that the previous version of the test is not clear regarding the `ChatChannel` creation, so explicitly deleting makes sense.